console/src/hardware/labjack/task/Write.tsx

```diff
@@ -0,0 +1,521 @@
+    schema: z.object({
+      name: z.string(),
+      config: writeTaskConfigZ,
+    }),
+  });
+
+  const [selectedChannels, setSelectedChannels] = useState<string[]>([]);
+  const [selectedChannelIndex, setSelectedChannelIndex] = useState<number | null>(null);
+
+  const taskState = useObserveState<WriteStateDetails>(
+    methods.setStatus,
+    methods.clearStatuses,
+    task?.key,
+    task?.state,
+  );
+
+  const createTask = useCreate<WriteTaskConfig, WriteStateDetails, WriteType>(
+    layoutKey,
+  );
+
+  const addStatus = Status.useAggregator();
+
+  const configure = useMutation({
+    mutationKey: [client?.key, "configure"],
+    onError: (e) => {
+      console.error(e);
+      addStatus({
+        variant: "error",
+        message: e.message,
+      });
+    },
+    mutationFn: async () => {
+      if (!(await methods.validateAsync()) || client == null) return;
+      const { name, config } = methods.value();
+
+      const dev = await client.hardware.devices.retrieve<Properties>(config.device);
+
+      let modified = false;
+      let shouldCreateStateIndex = primitiveIsZero(dev.properties.writeStateIndex);
+      if (!shouldCreateStateIndex) {
+        try {
+          await client.channels.retrieve(dev.properties.writeStateIndex);
+        } catch (e) {
+          if (NotFoundError.matches(e)) shouldCreateStateIndex = true;
+          else throw e;
+        }
+      }
+
+      if (shouldCreateStateIndex) {
+        modified = true;
+        const stateIndex = await client.channels.create({
+          name: `${dev.properties.identifier}_o_state_time`,
+          dataType: "timestamp",
+          isIndex: true,
+        });
+        dev.properties.writeStateIndex = stateIndex.key;
+        dev.properties.DO.channels = {};
+        dev.properties.AO.channels = {};
+      }
+
+      const commandsToCreate: WriteChan[] = [];
+      const statesToCreate: WriteChan[] = [];
+      for (const channel of config.channels) {
+        const key = `${channel.port}`;
+        const exPair = dev.properties[channel.type].channels[key];
+        if (exPair == null) {
+          commandsToCreate.push(channel);
+          statesToCreate.push(channel);
+        } else {
+          try {
+            await client.channels.retrieve([exPair.state]);
+          } catch (e) {
+            if (NotFoundError.matches(e)) statesToCreate.push(channel);
+            else throw e;
+          }
+          try {
+            await client.channels.retrieve([exPair.command]);
+          } catch (e) {
+            if (NotFoundError.matches(e)) commandsToCreate.push(channel);
+            else throw e;
+          }
+        }
+      }
+
+      if (statesToCreate.length > 0) {
+        modified = true;
+        const states = await client.channels.create(
+          statesToCreate.map((c) => ({
+            name: `${dev.properties.identifier}_${c.type}_${c.port}_state`,
+            index: dev.properties.writeStateIndex,
+            dataType: "uint8",
```

Comment:
@pjdotson The `dataType` of the state channel should be a `float32` if its an analog output, and a `uint8` if its a digital output